src/display/annotation_layer.js

```diff
@@ -697,7 +706,7 @@ var PopupAnnotationElement = (function PopupAnnotationElementClosure() {
 
       var selector = '[data-annotation-id="' + this.data.parentId + '"]';
       var parentElement = this.layer.querySelector(selector);
-      if (!parentElement) {
+      if (!parentElement || IGNORE_TYPES.indexOf(this.data.parentType) >= 0) {
```

Comment:
Nit: Perhaps a micro-optimization, but how about moving the `IGNORE_TYPE` check *before* doing `querySelector` to avoid a potentially useless element look-up?